src/acl/SourceDomain.h

```diff
@@ -23,6 +23,7 @@ class SourceDomainCheck: public ParameterizedNode< ACLData<const char *> >
 public:
     /* Acl::Node API */
     int match(ACLChecklist *) override;
+    bool requiresRequest() const override { return true; }
```

Comment:
I agree that both cases identified in PR description are buggy, but I am concerned about _this_ half of the fix: Prior to this PR, when there is no request, srcdomain "worked"[^1] with directives that do not support slow ACLs. With this change, srcdomain will always mismatch and log a warning. From UX point of view, that is a regression.

If possible, let's not requireRequest() for SourceDomainCheck but make recordLookup() call in LookupDone() conditional, with a level-3 "no request to recordLookup()" debugs() message for the `else` clause.

[^1]: Folks are not supposed to use that combination because it does not work reliably, and we do not have to apply heroic efforts to preserve that semi-working functionality, but some folks manage to use it successfully (from their point of view), and no heroic preservation efforts are required in this particular case AFAICT.

Reply:
Done at d1aaf26.

Reply:
Thank you. I have added the following sentence to PR description paragraph dedicated to srcdomain and srcdom_regex ACLs: This change does not start to require request presence for these two ACLs to avoid breaking any existing configurations that "work" without one.